crates/sui-graphql-rpc/src/types/move_registry/on_chain.rs

```diff
@@ -23,20 +24,14 @@ use crate::{
 
 use super::error::MoveRegistryError;
 
-const MAX_LABEL_LENGTH: usize = 63;
-
 /// Regex to parse a dot move name. Version is optional (defaults to latest).
 /// For versioned format, the expected format is `app@org/v1`.
 /// For an unversioned format, the expected format is `app@org`.
 ///
 /// The unbound regex can be used to search matches in a type tag.
 /// Use `VERSIONED_NAME_REGEX` for parsing a single name from a str.
-const VERSIONED_NAME_UNBOUND_REGEX: &str = concat!(
-    "([a-z0-9]+(?:-[a-z0-9]+)*)",
-    "@",
-    "([a-z0-9]+(?:-[a-z0-9]+)*)",
-    r"(?:\/v(\d+))?"
-);
+const VERSIONED_NAME_UNBOUND_REGEX: &str =
+    concat!(r"([a-z0-9.\-@]*)", r"\/", "([a-z0-9.-]*)", r"(?:\:v(\d+))?",);
```

Comment:
I don't follow why multiple `@`'s are allowed, vs having strictly one, but it's possible that's because I don't understand all the edge cases of the SuiNS domain name format. Here's the grammar I expected:

```
(?:
  ([a-z0-9.-]+\.)*
   [a-z0-9.-]+
)?            # Optional sub-names (may even not be there in the initial version)
@             # Separator
[a-z0-9.-]+   # Org name
/             # Separator
[a-z0-9.-]+   # App name
:v[1-9][0-9]* # Version
```

This may also negate the need for that extra parsing check you perform earlier (the one I had questions about).

Reply:
This also enables both SuiNS formats to work with this (`@org/app` and `org.sui/app`).

The goals were:
1. Add a more relaxed regex (to not have to enforce the SuiNS one here). 
2. Add validation for the NS part and the MVR (app) part separated on the parse stage.

Reply:
Understood and makes sense, thanks for explaining!